x-pack/plugins/siem/public/components/stat_items/barchart.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  // @ts-ignore
+  EuiSeriesChartUtils,
+} from '@elastic/eui';
+import { pure } from 'recompose';
+import styled from 'styled-components';
+// @ts-ignore
+import { EuiSeriesChart, EuiBarSeries, EuiXAxis, EuiYAxis } from '@elastic/eui/lib/experimental';
+import { BarChartData, WrappedByAutoSizer, ChartHolder } from '.';
+import { AutoSizer } from '../auto_sizer';
+
+const { SCALE, ORIENTATION } = EuiSeriesChartUtils;
+const getYaxis = (value: string | number) => {
+  const label = value.toString();
+  const labelLength = 4;
+  return label.length > labelLength ? `${label.slice(0, labelLength)}.` : label;
+};
+
+export const BarChartBaseComponent = pure<{
+  data: BarChartData[];
+  width: number | null | undefined;
+  height: number | null | undefined;
+}>(({ data, ...chartConfigs }) => {
+  return chartConfigs.width && chartConfigs.height ? (
+    // @ts-ignore
+    <SeriesChart
+      yType={SCALE.ORDINAL}
+      orientation={ORIENTATION.HORIZONTAL}
+      showDefaultAxis={false}
+      data-test-subj="stat-bar-chart"
+      {...chartConfigs}
+    >
+      {data.map(series => {
+        return (
+          // @ts-ignore
+          <EuiBarSeries
+            key={`stat-items-areachart-${series.key}`}
+            name={series.key}
+            // @ts-ignore
+            data={series.value!}
+            color={series.color}
+          />
+        );
+      })}
+      {/*
+// @ts-ignore */}
+      <EuiXAxis />
+      {/*
+// @ts-ignore */}
+      <EuiYAxis tickFormat={getYaxis} />
+    </SeriesChart>
+  ) : null;
+});
+
+export const BarChartWithCustomPrompt = pure<{
+  data: BarChartData[] | null | undefined;
+  height: number | null | undefined;
+  width: number | null | undefined;
+}>(({ data, height, width }) => {
+  return data &&
+    data.length &&
+    data.every(
+      ({ value }) => value != null && value.length > 0 && value.every(chart => chart.x != null)
+    ) ? (
+    <BarChartBaseComponent height={height} width={width} data={data} />
+  ) : (
+    <ChartHolder />
+  );
+});
+
+export const BarChart = pure<{ barChart: BarChartData[] | [] | null | undefined }>(
```

Comment:
Is the `[]` type needed here?

Tracing the usage of `BarChart` to `StatItemsComponent`, it looks like the type of `barChart` will only ever be `BarChartData[]` | `null` | `undefined`:

``` ts
export interface StatItemsProps extends StatItems {
  key: string;
  areaChart?: AreaChartData[];
  barChart?: BarChartData[];
}
```